salt/config/__init__.py

```diff
@@ -776,7 +779,7 @@
     'user': 'root',
     'root_dir': salt.syspaths.ROOT_DIR,
     'pki_dir': os.path.join(salt.syspaths.CONFIG_DIR, 'pki', 'minion'),
-    'id': None,
+    'id': '',
```

Comment:
This change appears to have broken masters and minions without IDs explicitely set. See #28339 and #28423 where I have reverted this line.